pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -279,17 +279,6 @@ def restore_optimizers(self) -> None:
 
         # restore the optimizers
         self.trainer.strategy.load_optimizer_state_dict(self._loaded_checkpoint)
-        for optimizer in self.trainer.optimizers:
-            # move optimizer to GPU 1 weight at a time
-            # avoids OOM
-            if self.trainer.root_gpu is not None:
-                for param, state in optimizer.state.items():
-                    if isinstance(state, dict):
-                        for k, v in state.items():
-                            if isinstance(v, torch.Tensor):
-                                state[k] = v.cuda(self.trainer.root_gpu)
-                    elif isinstance(state, torch.Tensor):
-                        optimizer.state[param] = state.cuda(self.trainer.root_gpu)
```

Comment:
@SeanNaren I see the DeepSpeed strategy overrides `load_optimizer_state_dict` to do nothing

After this change, do the optimizer states now need to be moved to the root_device within the strategy? Or does  the DeepSpeed engine internally already handle this?

Reply:
@ananthsub The test `test_deepspeed_multigpu_stage_3_warns_resume_training` should fail if the optimizer states where not moved to the root_device ? Can you make sure of this ?

Reply:
I had a brief look at the deepspeed code and didn't see anything that would break with these changes, however as @tchaton suggested I think the tests will verify this.